internal/gps/pkgtree/pkgtree_test.go

```diff
@@ -1775,128 +1744,194 @@ func TestFlattenReachMap(t *testing.T) {
+		return expect
 	}
 
-	// everything on
-	name = "simple"
-	except()
-	stdlib, main, tests = true, true, true
-	validate()
-
-	// turning off stdlib should cut most things, but we need to override the
-	// function
-	internal.IsStdLib = doIsStdLib
-	name = "no stdlib"
-	stdlib = false
-	except("encoding/binary", "go/parser", "hash", "net/http", "os", "sort")
-	validate()
-	// restore stdlib func override
-	overrideIsStdLib()
-
-	// stdlib back in; now exclude tests, which should just cut one
-	name = "no tests"
-	stdlib, tests = true, false
-	except("encoding/binary")
-	validate()
-
-	// Now skip main, which still just cuts out one
-	name = "no main"
-	main, tests = false, true
-	except("net/http")
-	validate()
-
-	// No test and no main, which should be additive
-	name = "no test, no main"
-	main, tests = false, false
-	except("net/http", "encoding/binary")
-	validate()
-
-	// now, the ignore tests. turn main and tests back on
-	main, tests = true, true
-
-	// start with non-matching
-	name = "non-matching ignore"
-	ignore = map[string]bool{
-		"nomatch": true,
-	}
-	except()
-	validate()
-
-	// should have the same effect as ignoring main
-	name = "ignore the root"
-	ignore = map[string]bool{
-		"github.com/example/varied": true,
-	}
-	except("net/http")
-	validate()
-
-	// now drop a more interesting one
-	name = "ignore simple"
-	ignore = map[string]bool{
-		"github.com/example/varied/simple": true,
-	}
-	// we get github.com/golang/dep/internal/gps from m1p, too, so it should still be there
-	except("go/parser")
-	validate()
-
-	// now drop two
-	name = "ignore simple and namemismatch"
-	ignore = map[string]bool{
-		"github.com/example/varied/simple":       true,
-		"github.com/example/varied/namemismatch": true,
-	}
-	except("go/parser", "github.com/Masterminds/semver")
-	validate()
-
-	// make sure tests and main play nice with ignore
-	name = "ignore simple and namemismatch, and no tests"
-	tests = false
-	except("go/parser", "github.com/Masterminds/semver", "encoding/binary")
-	validate()
-	name = "ignore simple and namemismatch, and no main"
-	main, tests = false, true
-	except("go/parser", "github.com/Masterminds/semver", "net/http")
-	validate()
-	name = "ignore simple and namemismatch, and no main or tests"
-	main, tests = false, false
-	except("go/parser", "github.com/Masterminds/semver", "net/http", "encoding/binary")
-	validate()
-
-	main, tests = true, true
-
-	// ignore two that should knock out gps
-	name = "ignore both importers"
-	ignore = map[string]bool{
-		"github.com/example/varied/simple": true,
-		"github.com/example/varied/m1p":    true,
-	}
-	except("sort", "github.com/golang/dep/internal/gps", "go/parser")
-	validate()
-
-	// finally, directly ignore some external packages
-	name = "ignore external"
-	ignore = map[string]bool{
-		"github.com/golang/dep/internal/gps": true,
-		"go/parser":                          true,
-		"sort":                               true,
+	for _, testCase := range []*flattenReachMapCase{
```

Comment:
So, just a note here...

I've been meaning to write an article about this pseudo-pattern for a while (I call them "transform table tests"), but haven't gotten around to it. This is a sloppier example (compare to e.g. `TestHashInputsOverrides()`, which actually looks like a table - though that one's also of a slightly different type), but the express purpose of the pattern is to avoid the sort of repetition in a table that makes it hard to tell, on visual scan, what's changing from one test to the next.

Structuring them up into a table reintroduces a bit of repetition - the boolean values don't carry through from one test to the next - but I'd say that's more than outweighed by the increase in readability that comes from structuring them into tables.

👍